### PR TITLE
Implementation of functionality for parsing a user provided CSV and creating user's and their chain of command stored in mongodb

### PR DESCRIPTION
**Pull request by austinhelton**
Implemented functionality for parsing a user provided CSV, creating user's and their chain of command, and storing them into their respective mongodb collection. This took around 12 hours between the initial creation, refactoring, and creating test cases.

Edge Case:
There is an interesting edge case for this though. When creating a user if they have a valid `manager_id` but the manager isn’t in the system. This is a situation where I would have gotten input from other team members on how to approach it. As I believe this design as some important impacts on the system. Below is my approach and what I believe another approach could be:
•My approach for this was to create a skeleton of the user, return its id, and let the process update the information later when it runs into that record. Unfortunately, this turns a create into an update and could be confusing to see those results as a user. 

•The alternative would be to fully create the user which would require refactoring the data processing into its on function (not too much of a lift). This would also require something along the lines of making the functionality to search for a user’s manager in the CSV recursive due to the following situation. Say you are creating a new user Bob and his manager Jim doesn’t exist yet. So, you go to fully create a user for Jim, but his manager Ashley isn’t in the system yet. This means we would need to follow that chain to the very last manager and then build backwards. This seemed like quite a bit of extra complexity. 

Chain of Command Creation:
I also decided that the creation of a user’s chain of command should be done after we process the CSV. This way we could do it in one go version trying to do it as we create/update them. As stated in the codes comment it could and probably should be optimized to not run through every user in the db by keep track of newly created records and which records has a manager update. The logic should be fairly simple and normally I would have done this (I love optimization) but I have been sick since starting this on Thursday and with the possible design changes above I wouldn’t want to implement more functionality until after discussing how to move forward with someone else. 

Requirements File:
I had an issue with `cffi==1.11.5` and `pkg-resources==0.0.0` but I am assuming this is because I done it on my person Windows machine. So, I did not change those in this commit, but I did add a package for the parsing of dates. I didn’t freeze the package version as I am not sure what version would work on Linux and I believe there is a slight difference in how Windows vs Linux does the package freeze. 

Anyways, thank you for the opportunity. This was fun besides being sick for the last few days. Let me know if you have any questions and I look forward to hearing from you.

### CODE REVIEW

File: src/test_handler.py [hunk 1]
```diff
@@ -55,7 +57,8 @@ def f():
 
         dummy_chain_of_commands = [
             {"user_id": dummy_users[0]["_id"], "chain_of_command":[]},
-            {"user_id": dummy_users[1]["_id"], "chain_of_command":[dummy_users[0]]},
+            {"user_id": dummy_users[1]["_id"],
+             "chain_of_command":[dummy_users[0]["_id"]]},
```
Comment (austinhelton) [inline src/test_handler.py:61]: I made this change to reflect what the readme says. Now it add just the user's id instead of the whole user to the chain of command
